Packs/ApiModules/Scripts/GetIncidentsApiModule/GetIncidentsApiModule.py

```diff
@@ -0,0 +1,228 @@
+        format_incident(inc, fields_to_populate, include_context)
+        for inc in incidents[:limit]
+    ]
+
+
+def prepare_fields_list(fields_list: list[str] | None) -> list[str]:
+    """Removes `incident.` prefix from the fields list and returns a list of unique values.
+
+    Args:
+        fields_list (list[str] | None): The current state of the fields list, as provided by the user.
+
+    Returns:
+        list[str]: The prepared fields list.
+    """
+    return list({
+        field.removeprefix("incident.") for field in fields_list if field
+    }) if fields_list else []
+
+
+def get_incidents(
+    custom_query: str | None = None,
+    incident_types: list[str] | None = None,
+    populate_fields: list[str] | None = None,
+    non_empty_fields: list[str] | None = None,
+    time_field: str = DEFAULT_TIME_FIELD,
+    from_date: datetime | None = None,
+    to_date: datetime | None = None,
+    include_context: bool = False,
+    limit: int = DEFAULT_LIMIT,
+    page_size: int = DEFAULT_PAGE_SIZE,
+) -> list[dict]:
+    """Performs a deeper formatting on the search arguments and runs paginated incidents search.
+
+    Args:
+        custom_query (str | None, optional): A custom query. Defaults to None.
+        incident_types (list[str] | None, optional): Incident types to retrieve. Defaults to None.
+        populate_fields (list[str] | None, optional): Incident fields to populate. Defaults to None.
+        non_empty_fields (list[str] | None, optional): Required non-empty incident fields. Defaults to None.
+        from_date (datetime | None, optional): The start date of the timeframe. Defaults to None.
+        to_date (datetime | None, optional): The end date of the timeframe. Defaults to None.
+        time_field (str, optional): The relevant time field to search by. Defaults to "created".
+        include_context (bool, optional): Whether or not to enrich the returned incidents with their the context data.
+            Defaults to False.
+        limit (int, optional): The search limit. Defaults to 500.
+        page_size (int, optional): Maximal page size. Defaults to 100.
+
+    Returns:
+        list[dict]: The requested incidents.
+    """
+    non_empty_fields = prepare_fields_list(non_empty_fields)
+
+    if populate_fields := prepare_fields_list(populate_fields):
+        populate_fields.extend(non_empty_fields + ["id"])
+        populate_fields = list(set(populate_fields))
+
+    query = build_query_parameter(
+        custom_query,
+        incident_types or [],
+        time_field,
+        from_date.isoformat() if from_date and time_field != "created" else None,
+        to_date.isoformat() if to_date and time_field != "created" else None,
+        non_empty_fields,
+    )
+
+    return get_incidents_with_pagination(
+        query,
+        from_date.astimezone().isoformat() if from_date and time_field == "created" else None,
+        to_date.astimezone().isoformat() if to_date and time_field == "created" else None,
```

Comment:
in the original implementation if `time_field == "created"` we pass to `getIncidents` the `from_date` and `to_date` values in both `query` argument and the dedicated fields, now we ensure we send it always once.